code/solr/SolrSearchAdapter.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+class SolrSearchAdapter implements SearchAdapterInterface
+{
+    /**
+     * @param SearchCriterion $criterion
+     * @return string
+     */
+    public function generateQueryString(SearchCriterion $criterion)
+    {
+        $writer = $this->getSearchQueryWriter($criterion);
+
+        return $writer->generateQueryString($criterion);
+    }
+
+    /**
+     * @param string $conjunction
+     * @return string
+     * @throws InvalidArgumentException
+     */
+    public function getConjunctionFor($conjunction)
+    {
+        switch ($conjunction) {
+            case SearchCriteria::UND:
+                return ' AND ';
+            case SearchCriteria::ODER:
+                return ' OR ';
+            default:
+                throw new InvalidArgumentException(
+                    sprintf('Invalid conjunction supplied to SolrSearchAdapter: "%s".', $conjunction)
+                );
+        }
+    }
+
+    /**
+     * @return string
+     */
+    public function getOpenComparisonContainer()
+    {
+        return '+(';
+    }
+
+    /**
+     * @return string
+     */
+    public function getCloseComparisonContainer()
+    {
+        return ')';
+    }
+
+    /**
+     * @param SearchCriterion $searchCriterion
+     * @return AbstractSearchQueryWriter
+     * @throws InvalidArgumentException
+     */
+    protected function getSearchQueryWriter(SearchCriterion $searchCriterion)
+    {
+        if ($searchCriterion->getSearchQueryWriter() instanceof AbstractSearchQueryWriter) {
+            // The user has defined their own SearchQueryWriter, so we should just return it.
+            return $searchCriterion->getSearchQueryWriter();
+        }
+
+        switch ($searchCriterion->getComparison()) {
+            case SearchCriterion::EQUAL:
+            case SearchCriterion::NOT_EQUAL:
+                return new SolrSearchQueryWriter_Basic();
+            case SearchCriterion::IN:
+            case SearchCriterion::NOT_IN:
+                return new SolrSearchQueryWriter_In();
```

Comment:
Can you please use injector for this instantiation, in case anyone wants to extend them?